### PR TITLE
bpo-11105: use a lower recursion limit for infinite recursion tests

### PR DESCRIPTION
**Pull request by isidentical**
https://bugs.python.org/issue11105

### CODE REVIEW

**Comment by isidentical** [comment]
@Fidget-Spinner does this patch prevent the crash for your end (do you have resources to test it out?). I highly doubt so (since these functions are giant I don't expect them to be inlined), though if it does then it would be good to know.

**Comment by Fidget-Spinner** [comment]
> @Fidget-Spinner does this patch prevent the crash for your end (do you have resources to test it out?). I highly doubt so (since these functions are giant I don't expect them to be inlined), though if it does then it would be good to know.

Thanks for working on this! Yup I can test it out, I use MSVC often.

Your doubts were right, this is sadly still causing a stack overflow. BTW, testing with buildbots should catch it, the ``AMD64 Windows10 Pro 3.x`` and ``AMD64 Windows10 3.x`` buildbots both caught it the previous commit. For PR pushes, their names are ``AMD64 Windows10 Pro PR`` and ``AMD64 Windows10 PR`` (the workers are ``bolen-windows10`` and ``kloth-win64``).

**Comment by isidentical** [comment]
> Both seem to have errored out though they haven't updated their statuses yet.

Cool! I was worried that they are not available on the public list, but apparently they are! Thanks for letting me know and testing this out.

**Comment by isidentical** [comment]
Seems like the error in `test_ast` doesn't happen anymore though we still get a stack overflow from somewhere else which doesn't make much sense since there is no ast compilation related cover on that test. 
```
test_recursion_in_except_handler (test.test_exceptions.ExceptionTests) ... 
D:\buildarea\3.x.bolen-windows10\build>exit /b -1073741571 
Windows fatal exception: stack overflow
Current thread 0x00000a54 (most recent call first):
  File "D:\buildarea\3.x.bolen-windows10\build\lib\test\test_exceptions.py", line 1132 in test_recursion_in_except_handler
  File "D:\buildarea\3.x.bolen-windows10\build\lib\unittest\case.py", line 549 in _callTestMethod
  File "D:\buildarea\3.x.bolen-windows10\build\lib\unittest\case.py", line 592 in run
  File "D:\buildarea\3.x.bolen-windows10\build\lib\unittest\case.py", line 652 in __call__
  File "D:\buildarea\3.x.bolen-windows10\build\lib\unittest\suite.py", line 122 in run
  File "D:\buildarea\3.x.bolen-windows10\build\lib\unittest\suite.py", line 84 in __call__
  File "D:\buildarea\3.x.bolen-windows10\build\lib\unittest\suite.py", line 122 in run
  File "D:\buildarea\3.x.bolen-windows10\build\lib\unittest\suite.py", line 84 in __call__
  File "D:\buildarea\3.x.bolen-windows10\build\lib\unittest\suite.py", line 122 in run
  File "D:\buildarea\3.x.bolen-windows10\build\lib\unittest\suite.py", line 84 in __call__
  File "D:\buildarea\3.x.bolen-windows10\build\lib\unittest\runner.py", line 176 in run
  File "D:\buildarea\3.x.bolen-windows10\build\lib\test\support\__init__.py", line 960 in _run_suite
  File "D:\buildarea\3.x.bolen-windows10\build\lib\test\support\__init__.py", line 1083 in run_unittest
  File "D:\buildarea\3.x.bolen-windows10\build\lib\test\libregrtest\runtest.py", line 210 in _test_module
  File "D:\buildarea\3.x.bolen-windows10\build\lib\test\libregrtest\runtest.py", line 246 in _runtest_inner2
  File "D:\buildarea\3.x.bolen-windows10\build\lib\test\libregrtest\runtest.py", line 282 in _runtest_inner
  File "D:\buildarea\3.x.bolen-windows10\build\lib\test\libregrtest\runtest.py", line 154 in _runtest
  File "D:\buildarea\3.x.bolen-windows10\build\lib\test\libregrtest\runtest.py", line 194 in runtest
  File "D:\buildarea\3.x.bolen-windows10\build\lib\test\libregrtest\main.py", line 321 in rerun_failed_tests
  File "D:\buildarea\3.x.bolen-windows10\build\lib\test\libregrtest\main.py", line 698 in _main
  File "D:\buildarea\3.x.bolen-windows10\build\lib\test\libregrtest\main.py", line 641 in main
  File "D:\buildarea\3.x.bolen-windows10\build\lib\test\libregrtest\main.py", line 719 in main
  File "D:\buildarea\3.x.bolen-windows10\build\lib\test\__main__.py", line 2 in <module>
  File "D:\buildarea\3.x.bolen-windows10\build\lib\runpy.py", line 86 in _run_code
  File "D:\buildarea\3.x.bolen-windows10\build\lib\runpy.py", line 196 in _run_module_as_main
```

**Comment by Fidget-Spinner** [comment]
> Seems like the error in `test_ast` doesn't happen anymore though we still get a stack overflow from somewhere else which doesn't make much sense since there is no ast compilation related cover on that test.
> 
> ```
> test_recursion_in_except_handler (test.test_exceptions.ExceptionTests) ... 


I just checked on main branch, this specific test also causes a stack overflow there. So maybe that test failure was previously masked by this one. Your patch fixes the AST stack overflow stack overflow issue for windows. Thanks!

Something I found strange is that the exceptions recursion test was added in https://github.com/python/cpython/commit/4e7a69bdb63a104587759d7784124492dcdd496e but it passed all along. I'm starting to suspect maybe there's some other change elsewhere that caused both `test_exception` and `test_ast` to fail. I'll start bisecting with `test_exception` soon. So please wait before merging this PR, thanks!

**Comment by Fidget-Spinner** [comment]
Ok, bisect tells me that this commit caused `test_exception` to fail:

```
commit f3fa63ec75fdbb4a08a10957a5c631bf0c4a5970
bpo-39573: Py_TYPE becomes a static inline function (GH-26493)

    Convert the Py_TYPE() and Py_SIZE() macros to static inline
    functions. The Py_SET_TYPE() and Py_SET_SIZE() functions must now be
    used to set an object type and size.
```

**Comment by Fidget-Spinner** [comment]
@isidentical I think if you rebase to the latest 6d518bb3a11f9b16098f45b21a13ebe8f537f045, test with buildbots should mostly pass. commit f3fa63ec75fdbb4a08a10957a5c631bf0c4a5970 was reverted.

File: Lib/test/support/__init__.py [hunk 1]
```diff
@@ -1999,3 +1999,12 @@ def check_disallow_instantiation(testcase, tp, *args, **kwds):
         qualname = f"{name}"
     msg = f"cannot create '{re.escape(qualname)}' instances"
     testcase.assertRaisesRegex(TypeError, msg, tp, *args, **kwds)
+
+@contextlib.contextmanager
+def infinite_recursion(max_depth=75):
```
Comment (Fidget-Spinner) [inline Lib/test/support/__init__.py:2004]: Minor suggestion: I think adding a docstring would help others know where/why/when they need this.

**Review by Fidget-Spinner — APPROVED** [review]
LGTM. The windows 10 tests are passing. Thanks for your efforts. Without your help I would've taken a lot longer to track down the other windows buildbot failures.

File: Lib/test/support/__init__.py (hunk 1 above)
Comment (isidentical) [reply Lib/test/support/__init__.py:2004]: Ah, missed this one! Yeap, forgot to do that. Will add a follow up PR.

**Pull request merged**